src/genworld_gui.cpp

```diff
@@ -833,7 +833,11 @@ struct GenerateLandscapeWindow : public Window {
 				break;
 
 			case WID_GL_BORDERS_RANDOM:
-				_settings_newgame.game_creation.water_borders = (_settings_newgame.game_creation.water_borders == BorderFlag::Random) ? BorderFlag{} : BorderFlag::Random;
+				if (_settings_newgame.game_creation.water_borders == BorderFlag::Random) {
+					_settings_newgame.game_creation.water_borders.Reset();
+				} else {
+					_settings_newgame.game_creation.water_borders = BorderFlag::Random;
+				}
```

Comment:
Hmm, would `_settings_newgame.game_creation.water_borders.Flip(BorderFlag::Random);` instead provide suitable behaviour? It would leave the water borders in whatever state they were, literally just toggling the Random flag.

Reply:
Somehow that breaks the button entirely, not allowing you to select Random unless all edges are freeform. Very strange.

All this goes away with #13289 anyway, which is a tri-state and changes the toggle button to a dropdown.